kernel/memory/src/paging/mod.rs

```diff
@@ -234,7 +234,7 @@ pub fn init(
     debug!("{:X?}\n{:X?}", aggregated_section_memory_bounds, _sections_memory_bounds);
     
     // bootstrap a PageTable from the currently-loaded page table
-    let current_active_p4 = frame_allocator::allocate_frames_at(aggregated_section_memory_bounds.page_table.start.1, 1)?;
+    let current_active_p4 = frame_allocator::allocate_frames_at(get_current_p4().start_address(), 1)?;
```

Comment:
while we're changing this, let's add a more detailed error since it's possible that the new page table `AllocatedFrame` might not be properly separated from the rest of the sections and therefore not available (this would indicate a build/link mistake on our part).
```suggestion
    let current_active_p4 = frame_allocator::allocate_frames_at(get_current_p4().start_address(), 1)
        .map_err(|_| "Failed to allocate frame for initial page table; is it merged with another section?")?;
```